tasks/components.py

```diff
@@ -0,0 +1,186 @@
+                print(f"** {file} defines a Component interface (bundles should not do so)")
+                ok = False
+                pass
+
+            path = file[: -len('/bundle.go')]
+            team = find_team(content)
+            doc = find_doc(content)
+
+            if team is None:
+                print(f"** {file} does not name a responsible team")
+                ok = False
+
+            bundles.append(Bundle(path, doc, team, []))
+
+    # assign components to bundles
+    bundles = [Bundle(b.path, b.doc, b.team, [c for c in components if c.path.startswith(b.path)]) for b in bundles]
+
+    # look for un-bundled components
+    for c in components:
+        if not any(c in b.components for b in bundles):
+            print(f"** component {c.path} is not in any bundle")
+            ok = False
+
+    return sorted(bundles), ok
+
+
+def make_components_md(bundles):
+    pkg_root = 'github.com/DataDog/dd-agent-comp-experiments/'
+    yield '# Agent Components'
+    yield ''
+    yield 'This file lists all components defined in this repository, with their package summary.'
+    yield 'Click the links for more documentation.'
+    yield ''
+    for b in bundles:
+        yield f'## [{b.path}](https://pkg.go.dev/{pkg_root}{b.path}) (Component Bundle)'
+        yield ''
+        yield f'*Datadog Team*: {b.team}'
+        yield ''
+        yield b.doc
+        for c in b.components:
+            yield f'### [{c.path}](https://pkg.go.dev/{pkg_root}{c.path})'
+            yield ''
+            if c.team != b.team:
+                yield f'*Datadog Team*: {c.team}'
+                yield ''
+            yield c.doc
+
+
+def make_codeowners(codeowners_lines, bundles):
+    codeowners_lines = codeowners_lines.__iter__()
+
+    # pass through the codeowners lines up to and including "# BEGIN COMPONENTS"
+    for line in codeowners_lines:
+        line = line.rstrip("\n")
+        yield line
+        if line == "# BEGIN COMPONENTS":
+            break
+
+    # codeowners is parsed in a last-match-wins fashion, so put more-specific values (components) after
+    # less-specific (bundles).  We include only components with a team different from their bundle, to
+    # keep the file short.
+    yield '/comp @DataDog/agent-shared-components'
+    different_components = []
+    for b in bundles:
+        if b.team:
+            yield f'/{b.path} @DataDog/{b.team}'
+        for c in b.components:
+            if c.team != b.team:
+                different_components.append(c)
+    for c in different_components:
+        if c.team:
+            yield f'/{c.path} @DataDog/{c.team}'
+
+    # drop lines from the existin codeowners until "# END COMPONENTS"
+    for line in codeowners_lines:
+        line = line.rstrip("\n")
+        if line == "# END COMPONENTS":
+            yield line
+            break
+
+    # pass through the rest of the file
+    for line in codeowners_lines:
+        yield line.rstrip("\n")
```

Comment:
(if you go with `splitlines`)
```suggestion
        yield line
```